### PR TITLE
Iam role policy changes check

### PR DESCRIPTION
**Pull request by pwyborn**
Why:
With reference to https://github.com/ministryofjustice/github-actions/pull/64
and [Create Github Action to identify PRs that contain IAM Role and Policy changes#3015](https://app.zenhub.com/workspaces/cloud-platform-team-5ccb0b8a81f66118c983c189/issues/ministryofjustice/cloud-platform/3015)

### CODE REVIEW

**Comment by pwyborn** [comment]
Mistake

File: .github/workflows/iam-role-policy-changes-check.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: Identify PRs that contain IAM Role and Policy changes
+on:
+  pull_request:
+    types: [opened, edited, reopened, synchronize]
+id: review_pr
```
Comment (poornima-krishnasamy) [inline .github/workflows/iam-role-policy-changes-check.yaml:5]:
```suggestion
```

File: .github/workflows/iam-role-policy-changes-check.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+env:
+   GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+jobs:
+  iam-role-policy-changes-check.yml:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+    runs-on: ubuntu-latest
```
Comment (poornima-krishnasamy) [inline .github/workflows/iam-role-policy-changes-check.yaml:15]: Duplicate runs-on. pls remove

File: .github/workflows/iam-role-policy-changes-check.yaml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    steps:
+      - uses: actions/checkout@master
```
Comment (poornima-krishnasamy) [inline .github/workflows/iam-role-policy-changes-check.yaml:17]: You will need something like
```
     - name: Checkout PR code
        uses: actions/checkout@master
     - name: Run iam/role policy changes check
        uses: ministryofjustice/github-actions/iam-role-policy-changes-check@iam-role-policy-changes-check
```

Comment (pwyborn) [reply .github/workflows/iam-role-policy-changes-check.yaml]: Hi @poornima-krishnasamy Poornima
Hopefully have corrected the above?

File: .github/workflows/iam-role-policy-changes-check.yaml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  iam-role-policy-changes-check.yml:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+    steps:
+      - name: Checkout PR code
+        uses: actions/checkout@master
+      - name: Run iam/role policy changes check
```
Comment (poornima-krishnasamy) [inline .github/workflows/iam-role-policy-changes-check.yaml:17]: Can u pls add the id here. It is used in the next step
```
id: review_pr
```

File: .github/workflows/iam-role-policy-changes-check.yaml [hunk 5]
```diff
@@ -0,0 +1,29 @@
+name: Identify PRs that contain IAM Role and Policy changes
+on:
+  pull_request:
+    types: [opened, edited, reopened, synchronize]
+env:
+   GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+jobs:
+  iam-role-policy-changes-check.yml:
```
Comment (poornima-krishnasamy) [inline .github/workflows/iam-role-policy-changes-check.yaml:8]: The identifier here cannot have ".".
```
iam-role-policy-changes-check
```

**Comment by pwyborn** [comment]
Tidying up - deleting branch and starting again

**Pull request closed without merging**